client/my-sites/post-share/index.jsx

```diff
@@ -42,55 +37,52 @@ import {
 	FEATURE_REPUBLICIZE,
 	PLAN_BUSINESS,
 } from 'lib/plans/constants';
-import Banner from 'components/banner';
-import Connection from './connection';
-import { isEnabled } from 'config';
 import SectionNav from 'components/section-nav';
 import NavTabs from 'components/section-nav/tabs';
 import NavItem from 'components/section-nav/item';
-import CompactCard from 'components/card/compact';
-import SocialLogo from 'social-logos';
-import EllipsisMenu from 'components/ellipsis-menu';
-import PopoverMenuItem from 'components/popover/menu-item';
+import Banner from 'components/banner';
+import Connection from './connection';
+import ActionsList from './publicize-actions-list';
+import { isEnabled } from 'config';
 import AsyncLoad from 'components/async-load';
 import formatCurrency from 'lib/format-currency';
+import {
+	FOOTER_SECTION_SCHEDULED,
+	FOOTER_SECTION_PUBLISHED,
+} from './constants';
 
 class PostShare extends Component {
 	static propTypes = {
-		siteSlug: PropTypes.string,
-		site: PropTypes.object,
-		post: PropTypes.object,
-		planSlug: PropTypes.string,
-		siteId: PropTypes.number,
-		postId: PropTypes.number,
-		isPublicizeEnabled: PropTypes.bool,
 		connections: PropTypes.array,
 		hasFetchedConnections: PropTypes.bool,
-		requestConnections: PropTypes.func
+		isPublicizeEnabled: PropTypes.bool,
+		planSlug: PropTypes.string,
+		post: PropTypes.object,
+		postId: PropTypes.number,
+		requestConnections: PropTypes.func,
+		site: PropTypes.object,
+		siteId: PropTypes.number,
+		siteSlug: PropTypes.string,
 	};
 
-	static FOOTER_SECTION_SCHEDULED = 'footer-section-scheduled';
-	static FOOTER_SECTION_PUBLISHED = 'footer-section-published';
+	static defaultProps = {
+		connections: [],
+	};
 
 	state = {
+		footerSection: FOOTER_SECTION_SCHEDULED,
 		message: PostMetadata.publicizeMessage( this.props.post ) || this.props.post.title,
 		skipped: PostMetadata.publicizeSkipped( this.props.post ) || [],
-		footerSection: PostShare.FOOTER_SECTION_SCHEDULED
-
 	};
 
 	setFooterSection = footerSection => () => this.setState( { footerSection } );
 
 	hasConnections() {
-		return !! ( this.props.connections && this.props.connections.length );
+		return !! this.props.connections.length;
```

Comment:
connections can be falsey I think, if you'd like to shorten this, let's use lodash get